presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsManager.java

```diff
@@ -75,7 +75,7 @@ public HistoryBasedPlanStatisticsCalculator getHistoryBasedPlanStatisticsCalcula
 
     public HistoryBasedPlanStatisticsTracker getHistoryBasedPlanStatisticsTracker()
     {
-        return new HistoryBasedPlanStatisticsTracker(() -> historyBasedPlanStatisticsProvider, historyBasedStatisticsCacheManager, sessionPropertyManager, config, isNativeExecution, serverVersion);
+        return new HistoryBasedPlanStatisticsTracker(() -> historyBasedPlanStatisticsProvider, historyBasedStatisticsCacheManager, planCanonicalInfoProvider, sessionPropertyManager, config, isNativeExecution, serverVersion);
```

Comment:
Here I need to pass the planCanonicalInfoProvider because it has getInputTableStatistics API. 

If we choose to skip input statistics check during optimization, we will have no input statistics when query completes. However we still need to track such statistics for other queries which choose to check input statistics. Hence this planCanonicalInfoProvider will be used to get the input statistics when writing histories back.